rhino-core/src/main/java/io/ryos/rhino/sdk/SimulationJobsScannerImpl.java

```diff
@@ -172,33 +171,44 @@ private Class getClassFor(String name) {
 
   private SimulationMetadata createBenchmarkJob(final Class clazz) {
 
+    // Simulation class.
     var simAnnotation = (io.ryos.rhino.sdk.annotations.Simulation) clazz
         .getDeclaredAnnotation(io.ryos.rhino.sdk.annotations.Simulation.class);
+
+    // Read runner annotation.
     var runnerAnnotation = (io.ryos.rhino.sdk.annotations.Runner) clazz
         .getDeclaredAnnotation(io.ryos.rhino.sdk.annotations.Runner.class);
+
+    // Read influx DB annotation, to enable influx db.
     var enableInflux = clazz.getDeclaredAnnotation(Influx.class) != null;
-    var stepMethods = Arrays.stream(clazz.getDeclaredMethods())
+
+    // Read scenario methods.
+    var scenarioMethods = Arrays.stream(clazz.getDeclaredMethods())
         .filter(m -> Arrays.stream(m.getDeclaredAnnotations())
             .anyMatch(a -> a instanceof io.ryos.rhino.sdk.annotations.Scenario))
         .map(s -> new Scenario(
             s.getDeclaredAnnotation(io.ryos.rhino.sdk.annotations.Scenario.class).name(), s))
         .collect(toList());
-    var specMethods = Arrays.stream(clazz.getDeclaredMethods())
-        .filter(m -> Arrays.stream(m.getDeclaredAnnotations())
-            .anyMatch(a -> a instanceof Dsl))
-        .map(s -> new Pair<>(s.getDeclaredAnnotation(Dsl.class).name(),
-            ReflectionUtils.<Spec>executeMethod(s, instanceOf(clazz).orElseThrow())))
+
+    // Create test instance.
+    var testInstance = instanceOf(clazz).orElseThrow();
+
+    var dsls = Arrays.stream(clazz.getDeclaredMethods())
+        .filter(method -> Arrays.stream(method.getDeclaredAnnotations())
+        .anyMatch(a -> a instanceof Dsl))
+        .map(s -> new Pair<>(s.getDeclaredAnnotation(Dsl.class).name(), ReflectionUtils.<LoadDsl>executeMethod(s, testInstance)))
         .map(p -> p.getSecond().withName(p.getFirst()))
         .collect(toList());
 
-    if (stepMethods.isEmpty() && isBlockingSimulation(runnerAnnotation)) {
+    if (scenarioMethods.isEmpty() && isBlockingSimulation(runnerAnnotation)) {
```

Comment:
at this level we do have DSLs or Scenarios, depending on the runner user selected.